### ISSUE TITLE
LSOA tiles colour scheme / MSOA zones

### ISSUE THREAD

**Issue by nikolai-b** [npct/pct-shiny#523]
The LSOA route network colour scheme legend is required to interpret the colours.  We currently have the MSOA zone colour legend so having another legend would be very confusing.  
We could hide the MSOA zones entirely when the user selects the LSOA route network but that is somewhat surprising to user: i.e. when the lines and routes are selected then the zones remain but when the LSOA route network is selected then the zones suddenly vanish which is against the principal of least surprise.
cc @mem48

**Comment by AnnaGoodman1** [comment]
that is a very good point Nikolai.  One option would be 1) modify the legend of the MSOA zones codes to be '% cycling to work (zones)', and then 2) Have a new legend titled 'No. cyclists on route (LSOA route network)' that only shows when the LSOA layer is chosen.  I think with the title to the zones clarified it might not be so confusing, what do you think @mem48 , @nikolai-b ? 

The other issue which comes up now that I see this in situ is that the experience of looking at the LSOA layer is made noticeably worse by having the outlines to the zones  on the screen - even when you uncheck 'show zones' you still have the zone boundaries there,  which look a bit like the lowest level of the route network. (Sorry, I had not thought of that when suggesting grey for the lowest level of cycling, but even if the lowest level were still pink I think the zone boundaries would make it much harder to take in the LSOA layer).  could the zone boundaries become transparent when someone unclicks 'show zones'? (@usr110 is that easy to do?).  I'm not sure they add much for the other layers, perhaps we could at least try it...

**Comment by mem48** [comment]
@AnnaGoodman1 I agree with your suggestion, there was no confusion with having two colour scales for the zones and the Index of Deprivation, and the code to show/hide the appropriate legend at the right time already exists. 

There shouldn't be any difficulty in removing the boundaries to the zones when clicking "Show Zones", the question is do we want to remove the zones completely when hidden or have them invisible but clickable? I would suggest invisible clickable objects are confusing to the user.

More general questions:

Do we want LSOA zones? Do we replace MSOA zones with LSOA or have both?
Do we want LSOA desire lines? Do we replace MSOA lines with LSOA or have both?

We can't have LSOA quietest routes because we don't have the data, and I don't think that shiny can handle that much data for the fastest routes, (the whole reason we have the raster LSOA in the first place).

Are we updating the Lines, Zones, and Model Output tabs with LSOA data?

Does the manual need updating?

**Comment by mem48** [comment]
There also seem to be a small bug, when changing the base map the LSOA raster will disappear

**Comment by Robinlovelace** [comment]
Regarding the overall question of how to fix the LSOA layer, it's important to realise what depends on what. The biggest blocker at present I think, now that the Go Dutch tiles have been generated (great work @mem48) is the colourscheme issue https://github.com/npct/pct-lsoa-vis/issues/2 - which I've just closed as I think everyone seems happy with the proposed solution now sat on the test server and looking good, e.g:

![image](https://cloud.githubusercontent.com/assets/1825120/22934777/13a992bc-f2c8-11e6-8f20-fff66ad1ee5f.png)

This issue, then is about any potential UI changes that need to be made to accommodate the additional layer. The options that have been proposed seem to have been proposed in this thread include:

- Modify the MSOA zone legend to say it relates to MSOAs and not LSOAs even when the LSOA route network is displayed and create a new legend for the LSOA route network (Anna).
- Making the zone boundaries invisible when the LSOA route network is shown (Anna).
- Add LSOA zones (Malcolm).
- Add LSOA desire lines (Malcolm).

Not wanting stifle new ideas but I'm against all these options. I think the zone legend title is fine as there is only one zone level legend and making it longer increases the minimum amount of visual information the user must process - to me the fact that it's at the MSOA is to me a detail that many people will not have heard of. I think making zone boundaries transparent is a good idea in general that will increase the visual appeal of the map when hiding zones. I'm in favour of that for all layers and would like to create a new issue: `Make zone boundaries invisible when Show Zones is off`.

Regarding how to deal with the duality between MSOA zone data and LSOA route network when the latter layer is on, I suggest simply defaulting to `Show Zones = FALSE` when that is the case. (I think we should also do that, and used to do that, with the IMD layer but that no longer seems to be the case, resulting in a rather chaotic mix of colours appearing when I clicked on the IMD layer as illustrated below):

![image](https://cloud.githubusercontent.com/assets/1825120/22936296/fae0a306-f2cc-11e6-9f4e-7ed82d5ed9a8.png)

This mess appears when you click on IMD ATM.

In summary, I think it should look like this, but without the zone boundaries and with a different legend, when the LSOA layer is selected:

![image](https://cloud.githubusercontent.com/assets/1825120/22936401/4f4eed80-f2cd-11e6-89bd-9e0de26ce58c.png)

**Comment by AnnaGoodman1** [comment]
I agree that better not to have the zones clickable when 'show zones' is not ticked, especially sincethey are not clickable when other types of lines are present.  so I suggest better in all scenarios to have the zones full invisible and non-clickable when 'show zones' not checked.

I think that, at least the moment, we should not have LSOA zones or the desire lines/routes (not least for data protection, for the latter).  Having zones + desire lines at the LSOA level would have been a good option if the user could make a binary LSOA/MSOA choice and then see everything being one or the other in a comparable way.  However since we can't offer the lines/routes at the LSOA level then I think it is better (at least for the time being) for us to have ‘mostly MSOA but a little LSOA’, as we currently do. 

Lines+Zones tab will be updated in due course through new data download page (which will include LSOA data). I think model output tab would need updating minimally under ‘Key Info’ to desc how LSOA layer created at the start, I can’t immediately see that a larger update is necessary elsewhere in tab (though suggestions welcome). You are right that we do need a new section for the manual, probably in practice LSOA layer can’t go live until that is there – not least as it could be confusing to have MSOA vs LSOA route  networks done in different ways – but DfT will I think/hope consider it done from a deliverable point of view as soon as rasters are on test server.

**Comment by AnnaGoodman1** [comment]
hi @Robinlovelace , your comment had not updated when I left mine so I had not seen it yet.  I think we agree on everything except that I think we should allow two legends simultaneously, one for the zone colours and one for the LSOA colours (and I don't think extending the zone legend to '% cycling to work (zones)' is that bad a lengthening?).  I also think it is fine toccontinue to show both colours/legendsfor the IMD, and let the user decide to deal with that - I agree It is not a lovely mapthat you get by doing that, but I think would probablybe more confusing/surprising to users if we started forcing the zones not show.

**Comment by AnnaGoodman1** [comment]
also I notice the IMD legend has a nice'show/hide' button, which allows the user to avoid visual clutter.  one option would be to introduce this show/hide function for the zones colour scheme and for the LSOA raster scheme, so that people can show/hide whichever legend they want?

**Comment by Robinlovelace** [comment]
I think that last suggestion, up here 👆, is excellent: 'dockable' legends are much more manageable. My preference would still be to turn the zones off by default when the LSOA route network is on to simplify the UI and the codebase. The user could then show the zones again, creating a 'colour overdose' if they really want to but I'd prefer that not to be the PCT's default behaviour.

**Comment by mem48** [comment]
To answer @AnnaGoodman1 questions on colours:

Yet we can change the colours, but it takes time as @nikolai-b has to rebuild the tiles for the whole country.

For the grey, we could choose a lighter colour, or make it slightly transparent which would have a similar effect. For the yellow, the choice of quite a pale yellow was to make it visually less emphasised than the higher values in green and light blue, the colour scheme currently has graduations in both intensity and colour. As yellow represents 10 - 50 cyclists then we really don't want it to stand out too much.

For the other colours, brighter is nor really an option (some of them are already at the brightest end of the spectrum, pink, green, cyan). I think the issue is that you want to see a whole city, but at that scale, the lines are very thin and thus hard to see.

I can think of two solutions.

1) LSOA Zones
2) Are more zoomed out levels for the LSOA route network we could resample the data to a lower resolution. E.g. rather than 10 x 10m pixles have 100 x 100 m pixles, which have the maximum consistent value. This would give you a "heat map" effect a bit like http://labs.strava.com/heatmap/#7/-1.49414/52.34876/yellow/bike

**Comment by mem48** [comment]
Another easy win would be to change the base map to something darker such as CartoDB.DarkMatter  https://leaflet-extras.github.io/leaflet-providers/preview/#filter=CartoDB.DarkMatter

**Comment by mem48** [comment]
A quick mockup of what a 100m aggregation would look like.
![100maggreagat](https://cloud.githubusercontent.com/assets/15819577/22977277/dac7eeb2-f385-11e6-8dac-d25c8feffbc6.png)

**Comment by Robinlovelace** [comment]
That looks good - I think auto switching to this 100m aggregated rasters below a certain zoom level, e.g. 8, shown below for Amsterdam, is a good plan.

![image](https://cloud.githubusercontent.com/assets/1825120/23029510/b5b52922-f462-11e6-9894-a8a359ba888d.png)

**Comment by Robinlovelace** [comment]
Back on the priority topic of UI, @mem48 and I are in agreement that the zone and route network colours should not simultaneously be on by default. Advantageous in terms of showing the user what's important quickly, which is the PCT's forte.

 @mem48 also raises the question of whether we should show rnet 1st by default. We've already discussed this and I have a mild preference for leaving it as is, with zones first as they are simpler and link nicely back to the national map but open to be persuaded otherwise given sufficiently strong arguments (e.g. most policy interventions are at the route network level, but stronger than that is needed for me to be persuaded.)

**Comment by mem48** [comment]
@Robinlovelace just thinking out loud. As the pan to change region system works, how about removing the home page and going straight to the full-screen map.  Showing the aggregated LSOA route network for the whole country. Then change the show zone behaviour to show MSOA when zoomed in and regions when zoomed out?

**Comment by mem48** [comment]
![varied vis](https://cloud.githubusercontent.com/assets/15819577/23072328/37fb548a-f529-11e6-94fb-5007253b08ab.png)

Quick demo of what the different aggregation at zoomed out levels will look like.

**Comment by usr110** [comment]
@npct/admin Just introduced an LSOA legend (based on the ranges and colours described in: https://github.com/npct/pct-lsoa-vis/blob/master/generate/colour.txt)

I wasn't sure when to display it. At the moment I am displaying it when the `Cycling Flows` is set to `Route Network (LSOA)` and also when the `Show Zones` is not selected. Should be straight-forward to change the mechanisms.

You can see it on the [test-server](http://npct0.vs.mythic-beasts.com/m/?r=cambridgeshire)

**Comment by mem48** [comment]
@usr110 Looks good, but the numbering is a bit odd can you condense down the ones with the same colour e.g. 1-9,10-49 rather than 1 , 2-9, 10, 11-49

**Comment by mem48** [comment]
Also, I think they LSOA ledgend should always show when the LSOA data is shown, even if that means having two legends at once.

**Comment by usr110** [comment]
@mem48 Sure. Please change the boundaries as you like here:

https://github.com/npct/pct-shiny/blob/master/regions_www/m/server.R#L38-L41

I used the boundaries and the colour values from the text file you supplied.

**Comment by mem48** [comment]
@usr110 I changed the bands and committed to master

**Comment by usr110** [comment]
great @mem48 !

**Comment by usr110** [comment]
Re https://github.com/npct/pct-shiny/issues/523#issuecomment-280695252 don't think `leaflet` allows multiple `legends`. We then need to use a panel for the LSOA legend. Let's see what others think about it.

**Comment by mem48** [comment]
@usr110  if it hard to do, then perhaps not top priority

**Comment by nikolai-b** [comment]
Great work @usr110 and @mem48.  I tweaked the text slightly and split it out from the zone legend so now it shows two legends.

**Comment by usr110** [comment]
Great additions @nikolai-b 
I didn't know that you could have layers even in legends.

**Comment by AnnaGoodman1** [comment]
@mem48 ,  thanks for the legend, but I think it is a mistake to have a colour for 'zero'?  
cc @nikolai-b

**Comment by usr110** [comment]
@AnnaGoodman1 Removed it from the legend.

**Comment by Robinlovelace** [comment]
Before it gets lost in the sands of time here's a preliminary answer to a good question by @mem48?

  >  how about removing the home page and going straight to the full-screen map. Showing the aggregated LSOA route network for the whole country. Then change the show zone behaviour to show MSOA when zoomed in and regions when zoomed out?

I think it's best to stick with the video and some words on the front page map. Listening to the video and reading some text is as far as many people will get.

**Comment by Robinlovelace** [comment]
Regarding the more specific issue of this thread, I'm very impressed with how the new LSOA rnet legend is looking, even when smothered by the area colours (you need a high res screen to accomodate both legends side-by-side):

![image](https://cloud.githubusercontent.com/assets/1825120/23129334/24281670-f77a-11e6-814d-d76f26057bd5.png)

Great work all.

**Comment by nikolai-b** [comment]
I think this issue is resolved, can we close it?

**Comment by usr110** [comment]
Indeed. Should be closed.
